llvm/lib/TableGen/Record.cpp

```diff
@@ -238,6 +238,12 @@ static void ProfileRecordRecTy(FoldingSetNodeID &ID,
     ID.AddPointer(R);
 }
 
+RecordRecTy::RecordRecTy(RecordKeeper &RK, ArrayRef<const Record *> Classes)
+    : RecTy(RecordRecTyKind, RK), NumClasses(Classes.size()) {
+  std::uninitialized_copy(Classes.begin(), Classes.end(),
```

Comment:
This can use a Range version in STLExtras.h. Will attempt that in a follow-on PR.

Reply:
Started draft https://github.com/llvm/llvm-project/pull/138174, but will wait for this one so I can update these uses as well.